### PR TITLE
Candy Machine Core: Upgrade Deps

### PR DESCRIPTION
**Pull request by samuelvanderwaal**

### CODE REVIEW

File: candy-machine-core/program/Cargo.toml [hunk 1]
```diff
@@ -18,12 +18,12 @@ cpi = ["no-entrypoint"]
...
 [dependencies]
-anchor-lang = "0.25.0"
+anchor-lang = "0.26.0"
 arrayref = "0.3.6"
-mpl-token-metadata = { version = "1.5.0", features = ["no-entrypoint"] }
-solana-program = "1.10.29"
-spl-associated-token-account = { version = "~1.0.5", features = ["no-entrypoint"] }
-spl-token = { version = "~3.3.1", features = ["no-entrypoint"] }
+mpl-token-metadata = { version = "1.7.0", features = ["no-entrypoint"] }
+solana-program = "1.14.13"
```
Comment (febo) [inline candy-machine-core/program/Cargo.toml:24]: Do we want to have this as `1.14`? It seems that the majority of other projects are using `1.14`.

Comment (samuelvanderwaal) [reply candy-machine-core/program/Cargo.toml:24]: Yeah, good point.

**Review by blockiosaurus — APPROVED** [review]
This PR is updating the Anchor and Solana versions to the latest for Candy Machine Core. It also makes a couple minor tweaks and calls the latest create_metadata_accounts call.

**Review by febo — APPROVED** [review]
LGTM

**Pull request merged**